hedera-node/src/main/java/com/hedera/services/context/NodeInfo.java

```diff
@@ -61,14 +73,74 @@ public boolean isZeroStake(long nodeId) {
 		return isZeroStake[index];
 	}
 
-	private void readBook() {
+	/**
+	 * Convenience method to check if this node is zero-stake.
+	 */
+	public boolean isSelfZeroStake() {
+		return isZeroStake(selfId);
+	}
+
+	/**
+	 * Returns the account parsed from the address book memo corresponding
+	 * to the given node id.
+	 *
+	 * @param nodeId the id of interest
+	 * @throws IllegalArgumentException if the book did not contain the id, or was missing an account for the id
+	 */
+	public AccountID accountOf(long nodeId) {
+		if (!bookIsRead) {
+			readBook();
+		}
+
+		final int index = (int)nodeId;
+		if (index < 0 || index >= numberOfNodes) {
+			throw new IllegalArgumentException("No node with id " + nodeId + " was in the address book!");
+		}
+		final var account = accounts[index];
+		if (account == null) {
+			throw new IllegalArgumentException("The  address book did not have an account for node id " + nodeId + "!");
```

Comment:
nit: two spaces before `address book`. Could have been caught if the test also expects the exception error message. Should add that expectation in the test.

Reply:
Fixed typo.